File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/BrowseMemberRequest.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.structured.BrowseDescription;
import org.opcfoundation.webserver.types.common.UaObjectId;

public class BrowseMemberRequest {
    private final UaObjectId objectId;
    private final String childId;
    private final boolean isMethod;
    private final BrowseDescription browseDescription;

    public BrowseMemberRequest(
            UaObjectId objectId,
            String childId,
            boolean isMethod,
            BrowseDescription browseDescription)
    {
        this.objectId = objectId;
        this.childId = childId;
        this.isMethod = isMethod;
        this.browseDescription = browseDescription;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public String getChildId() {
        return childId;
    }

    public boolean isMethod() {
        return isMethod;
    }

    public BrowseDescription getBrowseDescription() { return browseDescription; }
}

File: opcua-webservice-node/source/src/server/service/message/reactiveobject/BrowseMemberRequest.ts
import { UaBrowseDescription } from "opcua-webapi-ts";
import { UaObjectId } from "../../../types";

export class BrowseMemberRequest {
    private readonly _objectId: UaObjectId;
    private readonly _childId: string;
    private readonly _browseDescription: UaBrowseDescription;

    constructor(
        objectId: UaObjectId,
        childId: string,
        browseDescription: UaBrowseDescription,
    ) {
        this._objectId = objectId;
        this._childId = childId;
        this._browseDescription = browseDescription;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get childId(): string {
        return this._childId;
    }

    get browseDescription(): UaBrowseDescription {
        return this._browseDescription;
    }
}